zarr/core.py

```diff
@@ -1854,10 +1872,12 @@ def _process_chunk(
                 # contiguous, so we can decompress directly from the chunk
                 # into the destination array
                 if self._compressor:
-                    if isinstance(cdata, PartialReadBuffer):
+                    if isinstance(cdata, (PartialReadBuffer, UncompressedPartialReadBufferV3)):
```

Comment:
How does one end up with UncompressedPartial here? Isn't that what happens simply if there is no compression?